ValheimPlus/GameClasses/Fireplace.cs

```diff
@@ -40,20 +44,67 @@ private static void Postfix(ref double __result, ref Fireplace __instance)
             {
                 if (!Configuration.Current.FireSource.IsEnabled) return;
 
-                if (Configuration.Current.FireSource.onlyTorches)
+                if (FireplaceExtensions.IsTorch(__instance.m_name))
                 {
-                    // if configuration is set to only keep torches lit, check that our current instance is a torch and only then intercept and overwrite result
-                    if (FireplaceExtensions.IsTorch(__instance.m_name))
+                    if (Configuration.Current.FireSource.torches)
                     {
                         __result = 0.0;
                     }
                 }
-                else
+                else if (Configuration.Current.FireSource.fires)
                 {
                     __result = 0.0;
                 }
             }
         }
+
+        [HarmonyPatch(typeof(Fireplace), nameof(Fireplace.UpdateFireplace))]
+        public static class Fireplace_UpdateFireplace_Transpiler
+        {
+            private static MethodInfo method_Fireplace_UpdateState = AccessTools.Method(typeof(Fireplace), nameof(Fireplace.UpdateState));
+            private static MethodInfo method_addFuelFromNearbyChests = AccessTools.Method(typeof(Fireplace_UpdateFireplace_Transpiler), nameof(Fireplace_UpdateFireplace_Transpiler.addFuelFromNearbyChests));
+
+            [HarmonyTranspiler]
+            public static IEnumerable<CodeInstruction> Transpile(IEnumerable<CodeInstruction> instructions)
+            {
+                if (!Configuration.Current.FireSource.IsEnabled || !Configuration.Current.FireSource.autoFuel) return instructions;
+
+                List<CodeInstruction> il = instructions.ToList();
+
+                for (int i = 0; i < il.Count; i++)
+                {
+                    if (il[i].Calls(method_Fireplace_UpdateState))
+                    {
+                        il.Insert(++i, new CodeInstruction(OpCodes.Ldarg_0));
+                        il.Insert(++i, new CodeInstruction(OpCodes.Call, method_addFuelFromNearbyChests));
```

Comment:
Not inside IsOwner check

Should be after instance `bool ZNetView::IsOwner()` before the code check for the current value of the fuel.

Reply:
Corrected!

_PS: It stings knowing that I was the one who told you that on discord 😂_

Reply:
We did come up with the fix together though 😛